app/routes.py

```diff
@@ -1,2 +1,286 @@
+                })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if request_body.get("title") and request_body.get("description"): 
+            if request_body.get("completed_at")==None:
+                new_task = Task(title=request_body["title"],
+                                description=request_body["description"])
+                db.session.add(new_task)
+                db.session.commit()
+                response_value = {"task":{
+                        "id": new_task.task_id,
+                        "title": new_task.title,
+                        "description": new_task.description,
+                        "is_complete": False
+                }}
+            elif request_body.get("completed_at"):
+                new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=datetime.now())
+                db.session.add(new_task)
+                db.session.commit()
+                response_value = {"task":{
+                        "id": new_task.task_id,
+                        "title": new_task.title,
+                        "description": new_task.description,
+                        "is_complete": True
+                    }}
+        else:
+            return make_response({"details":"Invalid data"}, 400)
+
+        return make_response(response_value, 201)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        if not task.completed_at:
+            if task.goal_id:
+                response_value = {"task":{
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                }}
+            else:
+                response_value = {"task":{
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                }}
+
+            return make_response(response_value, 200)
+        else:
+            response_value = {"task":{
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at
+            }}
+
+            return make_response(response_value, 200)
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        if form_data.get("completed_at"):
+            task.completed_at = datetime.now()
```

Comment:
Ooo this is also a way to mark a task as complete. However, this logic seems to override the value of `completed_at` in the request body with the current time. Was this intentional?